### ISSUE TITLE
Crash and Dump file

### ISSUE THREAD

**Issue by JBloodyCut** [OpenRCT2/OpenRCT2#4410]
**OS:** Windows 7
**Version:**  0.0.5
**Commit/Build:** b0a8fb0

I was playing RCT1 "Dynamite Dunes" scenario.. (OPEN RCT2)

At end of scenario(oct 31 3year) , stop the screen and crash & error occured

please fix it.. i want to complete this scenario.. 

Thank you.. sorry for the bad english..
- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Multiplayer?

**Dump file**

[eaaef2a0-e5bc-480f-9c06-8a42da4be648(b0a8fb0).zip](https://github.com/OpenRCT2/OpenRCT2/files/466009/eaaef2a0-e5bc-480f-9c06-8a42da4be648.b0a8fb0.zip)

**Screenshots / Video:**

Sorry.. because of crash, i cant capture it..

![error1](https://cloud.githubusercontent.com/assets/22131518/18417662/316ffbdc-7871-11e6-8ae8-cb1aac2227ac.jpg)

**Save game:**

[eaaef2a0-e5bc-480f-9c06-8a42da4be648.zip](https://github.com/OpenRCT2/OpenRCT2/files/466010/eaaef2a0-e5bc-480f-9c06-8a42da4be648.zip)

[Dynamite Dunes2.zip](https://github.com/OpenRCT2/OpenRCT2/files/466011/Dynamite.Dunes2.zip)

**Comment by Broxzier** [comment]
You forgot to include the dump file in the zip, you only uploaded the saved games. Also please fix up this issue, it's full of placeholders that you left in.

**Comment by janisozaur** [comment]
It looks like the dump is attached

**Comment by Gymnasiast** [comment]
Possibly fixed by https://github.com/OpenRCT2/OpenRCT2/commit/0ed526bb0d25cf93d9634028505b9cc30474221d, but we wouldn't know for sure without examing the dump.

**Comment by duncanspumpkin** [comment]
Sounds like the sprite limit getting hit. Which has also been fixed.

**Comment by scheromat** [comment]
the same issue here - at end of scenario(oct 31 3year) , stop the screen and crash & error occure
[autosave_2016-09-25_21-52-36.zip](https://github.com/OpenRCT2/OpenRCT2/files/491867/autosave_2016-09-25_21-52-36.zip)
d

[dump1_3f1a52f0-2eb2-4c45-b37e-5e7d3f2be7a7(d953f25).zip](https://github.com/OpenRCT2/OpenRCT2/files/491869/dump1_3f1a52f0-2eb2-4c45-b37e-5e7d3f2be7a7.d953f25.zip)
[dump2_498523a8-caf1-4928-9ac7-08512ac91234.zip](https://github.com/OpenRCT2/OpenRCT2/files/491870/dump2_498523a8-caf1-4928-9ac7-08512ac91234.zip)

**Comment by Nubbie** [comment]
From [dump1_3f1a52f0-2eb2-4c45-b37e-5e7d3f2be7a7(d953f25).zip](https://github.com/OpenRCT2/OpenRCT2/files/491869/dump1_3f1a52f0-2eb2-4c45-b37e-5e7d3f2be7a7.d953f25.zip)

**Duplicate of** #2825 , #4644 - Fixed? (seems like it in #4644)

**Error message;**
Unhandled exception thrown: write access violation.
steam was nullptr.

**Call stack;**
```
>	openrct2.dll!steam_particle_create(short x, short y, short z) Line 6649	C
 	openrct2.dll!sub_6D63D4(rct_vehicle * vehicle=0x011105bc) Line 6687	C
 	openrct2.dll!vehicle_update_track_motion(rct_vehicle * vehicle, int * outStation=0x00000000) Line 8616	C
 	openrct2.dll!vehicle_update_travelling(rct_vehicle * vehicle=0x011105bc) Line 3170	C
 	openrct2.dll!vehicle_update(rct_vehicle * vehicle=0x011105bc) Line 1668	C
 	openrct2.dll!vehicle_update_all(...) Line 1058	C
 	openrct2.dll!game_logic_update(...) Line 370	C
 	openrct2.dll!game_update(...) Line 292	C
 	openrct2.dll!rct2_update(...) Line 468	C
 	openrct2.dll!openrct2_loop(...) Line 468	C
 	openrct2.dll!openrct2_launch(...) Line 342	C
 	openrct2.dll!StartOpenRCT(HINSTANCE__ * hInstance=0x00400000, HINSTANCE__ * hPrevInstance=0x00000000, char * lpCmdLine=0x016640ab, int nCmdShow=1) Line 134	C
 	openrct2.exe!004069e4()	Unknown
 	[Frames below may be incorrect and/or missing, no symbols loaded for openrct2.exe]	
 	[External Code]	
```

**Comment by janisozaur** [comment]
Fixed in https://github.com/OpenRCT2/OpenRCT2/commit/82b18ddaf371fe30f648be7bedf81afa56ebffa9
